pilot/pkg/config/aggregate/config.go

```diff
@@ -59,24 +59,24 @@ func makeStore(stores []model.ConfigStore, writer model.ConfigStore) (model.Conf
 
 // MakeWriteableCache creates an aggregate config store cache from several config store caches. An additional
 // `writer` config store is passed, which may or may not be part of `caches`.
-func MakeWriteableCache(caches []model.ConfigStoreController, writer model.ConfigStore) (model.ConfigStoreController, error) {
+func MakeWriteableCache(caches []model.ConfigStoreController, writer model.ConfigStore) model.ConfigStoreController {
 	stores := make([]model.ConfigStore, 0, len(caches))
 	for _, cache := range caches {
 		stores = append(stores, cache)
 	}
 	store, err := makeStore(stores, writer)
```

Comment:
```suggestion
	store, err := makeStore(stores, writer)
```

Nit: might as well go one level deeper here and drop `err` from `makeStore` - it always returns `nil` on err anyway.

Reply:
tbh this was just a miss here - didn't realize I was ignoring the error here. But it can only fail if schemas.Validate fails which is a bit silly since its all static anyways, so I will remove that call

Reply:
actually ill just make it a debug assert